configuration/azure/appconfig/appconfig.go

```diff
@@ -248,9 +262,75 @@ func (r *ConfigurationStore) getLabelFromMetadata(metadata map[string]string) *s
 }
 
 func (r *ConfigurationStore) Subscribe(ctx context.Context, req *configuration.SubscribeRequest, handler configuration.UpdateHandler) (string, error) {
-	return "", fmt.Errorf("Subscribe is not implemented by this configuration store")
+	sentinelKey := r.getSentinelKeyFromMetadata(req.Metadata)
+	if sentinelKey == "" {
+		return "", fmt.Errorf("sentinel key is not provided in metadata")
+	}
+	uuid, err := uuid.NewRandom()
+	if err != nil {
+		return "", fmt.Errorf("failed to generate subscription id, error is %w", err)
+	}
+	subscribeID := uuid.String()
+	stop := make(chan struct{})
+	r.subscribeStopChanMap.Store(subscribeID, stop)
+	go r.doSubscribe(ctx, req, handler, sentinelKey, subscribeID, stop)
+	return subscribeID, nil
+}
+
+func (r *ConfigurationStore) doSubscribe(ctx context.Context, req *configuration.SubscribeRequest, handler configuration.UpdateHandler, sentinelKey string, id string, stop chan struct{}) {
+	for {
+		// get sentinel key changes
+		_, err := r.Get(ctx, &configuration.GetRequest{
+			Keys:     []string{sentinelKey},
+			Metadata: req.Metadata,
+		})
+		if err != nil {
+			r.logger.Debugf("fail to get sentinel key changes or sentinel key's value is unchanged: %s", err)
+		} else {
+			items, err := r.Get(ctx, &configuration.GetRequest{
```

Comment:
Needs a timeout here. Use a sub-context with a timeout

Reply:
Instead of having timeout here I think we shoud have it in actual Get #191. 
But, While testing I found that when a cancelled context is passed to GetSetting#191, az appconfig sdk panics.
refer - https://github.com/Azure/azure-sdk-for-go/issues/19223
For now, I have used context.TODO in the actual GetSettings call. We need to change this when there is fix in sdk.

Reply:
Sounds good. Can you please put a comment with a TODO for remind us all in the future about this? Feel free to link to the issue in the SDK repo.